tests/unit/package_managers/test_pip.py

```diff
@@ -3377,33 +3363,90 @@ def test_download_dependencies(
         ) in caplog.text
         # </check basic logging output>
 
+        # <check downloaded wheels>
+        if allow_binary:
+            assert f"Downloading {len(wheels)} wheel(s) ..." in caplog.text
+            # wheel #2 does not match any checksums
+            assert "The file is removed from the output directory" in caplog.text
+        # </check downloaded wheels>
```

Comment:
can we test that we remove the one we intended to remove?

Reply:
I do not know how to create the file files before calling `download_dependencies`, since if a file already exists, it will not be downloaded again, so `path.unlink()` will not be called :confused:

Reply:
IMO it would be enough to log the filename and check that the log message includes the right filename

We can test the actual behavior with integration tests later